src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackProcessor.java

```diff
@@ -96,13 +112,24 @@ private void attachCase(String exceptionRecordJurisdiction,
         return emptyList();
     }
 
-    @SuppressWarnings({"unchecked", "squid:S1135"})
-    //TODO WIP
     private Map<String, Object> insertNewScannedDocument(List<Map<String, Object>> exceptionDocuments,
-                                                         Map<String, Object> caseData) {
-        //TODO check that document Id is unique and not duplicate in caseData
-        List<Object> caseList = (List<Object>) caseData.get(SCANNED_DOCUMENTS);
-        caseList.addAll(exceptionDocuments);
-        return ImmutableMap.of(SCANNED_DOCUMENTS, caseList);
+                                                         List<Map<String, Object>> existingDocuments) {
+        //TODO assert that there is one document in the exception record.
```

Comment:
is 1 document correct?

Reply:
it's not correct exception record can have multiple documents

Reply:
Ok sure ... Il change that. Its was an assumption. 
That will make the duplicate document check a bit more complex

Reply:
So the correct behaviour on the detection of a duplicate document 1 or more via document number in the case of multiple documents should be ?
1 - error and do not progress the attachment
2 - Leave out the duplicates ( what about newer or changed document) ?
3 - put a warning about duplicate documents and do 2 ?

Reply:
Note CCD will fail if there is a duplicate DCN and do so quietly to the user.

Reply:
error I guess? is it likely to happen? i.e. is there a legitimate way that it's possible to happen?

Reply:
if you attach the same exception record twice

Reply:
The way i came across it is attaching a duplicate exception record ( same message sent twice) and trying to attach it to the same case.

Reply:
Ill add this to a new PR#14 to deal with multiple received documents. Which I'll complete once this is merged.

Reply:
Ill update the //TODO comment

Reply:
yes that is exactly why we need to do the duplicate check.
Hence this PR